rxandroidble/src/test/groovy/com/polidea/rxandroidble/RxBleClientTest.groovy

```diff
@@ -79,6 +79,49 @@ class RxBleClientTest extends Specification {
         1 * bleAdapterWrapperSpy.startLeScan(_) >> true
     }
 
+    def "should start BLE scan if bluetooth is enabled before subscribing to the scan observable"() {
+        given:
+        TestSubscriber testSubscriber = new TestSubscriber<>()
+        bleAdapterWrapperSpy.hasBluetoothAdapter() >> true
+        bleAdapterWrapperSpy.isBluetoothEnabled() >> true
+
+        when:
+        def scanObservable = objectUnderTest.scanBleDevices(null)
+
+        then:
+        0 * bleAdapterWrapperSpy.isBluetoothEnabled()
+
+        when:
+        scanObservable.subscribe(testSubscriber)
+
+        then:
+        1 * bleAdapterWrapperSpy.startLeScan(_) >> true
+    }
+
+    def "should start scan for only subscribers with no exceptions from scan observable"() {
```

Comment:
Could you rephrase it to be easier to understand? Maybe something like: `"should check if /condition/ is met at the time of each subscription"`
It could like like this: 
```
        given:
        def firstSubscriber = new TestSubscriber<>()
        def secondSubscriber = new TestSubscriber<>()
        bleAdapterWrapperSpy.hasBluetoothAdapter() >> true
        bleAdapterWrapperSpy.startLeScan(_) >> true
        def scanObservable = objectUnderTest.scanBleDevices(null)

        when:
        scanObservable.subscribe(firstSubscriber)

        then:
        1 * bleAdapterWrapperSpy.isBluetoothEnabled() >> false
        
        and:
        firstSubscriber.assertError {
            BleScanException exception -> exception.reason == BLUETOOTH_DISABLED
        }

        when:
        scanObservable.subscribe(secondSubscriber)

        then:
        1 * bleAdapterWrapperSpy.isBluetoothEnabled(_) >> true
        
        and:
        secondSubscriber.assertNoErrors()
```
Then we would need to make the test exercise for other conditions (`isLocationPermissionOk` / `isLocationProviderOk`). It could be quite easily achieved by using `where:` clause